Add request validation

CrazyTapok opened the pull request:
1. Screenshot: 
![image](https://github.com/user-attachments/assets/baf50883-5eb3-4ec0-906c-e81f921ab064)


2. What has been implemented: 
  - [x] add request validation for models and configuring their
  - [x] add ValidationServiceExtensions for configuring FluentValidation and registration of all validators

CrazyTapok committed:
feat: Add request validation
add request validation for models and configuring their, add ValidationServiceExtensions for configuring FluentValidation and registration of all validators.

svirzhevskiy commented on src/TaskManager.API/Contracts/Extensions/ValidationServiceExtensions.cs:
```diff
@@ -0,0 +1,23 @@
+        services.AddFluentValidationClientsideAdapters();
+
+        // Registration of all validators
+        services.AddValidatorsFromAssemblyContaining<CompanyRequestValidator>();
```
оно добавляет валидпторы из сборки содержащие указанный класс. У тебя все валидаторы в одной сборке, соответственно последующие вызовы ничего не делают

CrazyTapok replied:
Разобрался, спасибо.

svirzhevskiy commented on src/TaskManager.API/Contracts/Validators/CompanyRequestValidator.cs:
```diff
@@ -0,0 +1,14 @@
+    public CompanyRequestValidator()
+    {
+        RuleFor(company => company.Title)
+            .NotEmpty().WithMessage("The company name is required.")
```
а разве там нет почти такого ще дефолтного месаджа? Протести, по-моему там такой же дефолтный

CrazyTapok replied:
Ты прав, есть.

svirzhevskiy commented on src/TaskManager.API/Contracts/Validators/TaskRequestValidator.cs:
```diff
@@ -0,0 +1,36 @@
+        RuleFor(task => task.CreatedDate)
+            .LessThanOrEqualTo(DateTime.UtcNow).WithMessage("The creation date cannot be in the future.");
+
+        RuleFor(task => task.UpdatedDate)
```
тут можешь добавить, что updated date не может быть меньше created date, заодно посмотришь как два проперти между собой валидировать

CrazyTapok committed:
refactor: fix remarks after the review
Fix class ValidationServiceExtensions. In validators, some custom messages have been replaced with default ones.  Add English culture.
